### PR TITLE
DBZ-228 Fix MySQL connector unsigned integer types

### PR DESCRIPTION
**Pull request by omarsmak**
Reference issue: https://issues.jboss.org/browse/DBZ-228
This PR will fix the unsigned issue for the following integer types: **TINYINT, SMALLINT, MEDIUMINT, INT, BIGINT**. 
Note: I was using [this reference](https://kafka.apache.org/0102/javadoc/org/apache/kafka/connect/data/Schema.Type.html) for the schema mapping.
The followings are the mapping of the Literal type:

- `TINYINT` -> `INT16` 
- `TINYINT UNSIGNED` -> `INT32` 
- `SMALLINT` -> `INT16` 
- `SMALLINT UNSIGNED` -> `INT32` 
- `MEDIUMINT` -> `INT32` 
- `MEDIUMINT UNSIGNED` -> `INT32` 
- `INT` -> `INT32` 
- `INT UNSIGNED` -> `INT64` 
- `BIGINT` -> `INT64`
- `BIGINT UNSIGNED` -> `org.apache.kafka.connect.data.Decimal:Byte`

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java [hunk 1]
```diff
@@ -43,29 +44,32 @@
 /**
  * MySQL-specific customization of the conversions from JDBC values obtained from the MySQL binlog client library.
  * <p>
- * This class always uses UTC for the default time zone when converting values without timezone information to values that require
+ * This class always uses UTC for the default time zone when converting values without timezone information to values
+ * that require
```
Comment (rhauch) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlValueConverters.java]: There are too many unnecessary formatting changes in this diff, and it's impossible to easily see what changes are important for this fix. Can you revert most of these?

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorRegressionIT.java [hunk 2]
```diff
@@ -71,20 +71,20 @@ public void afterEach() {
...
+                .with(MySqlConnectorConfig.SSL_MODE, SecureConnectionMode.DISABLED)
+                .with(MySqlConnectorConfig.SERVER_ID, 18765)
+                .with(MySqlConnectorConfig.SERVER_NAME, "regression")
+                .with(MySqlConnectorConfig.POLL_INTERVAL_MS, 10)
+                .with(MySqlConnectorConfig.DATABASE_WHITELIST, "regression_test")
+                .with(MySqlConnectorConfig.DATABASE_HISTORY, FileDatabaseHistory.class)
+                .with(MySqlConnectorConfig.INCLUDE_SCHEMA_CHANGES, true)
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, SnapshotMode.NEVER.toString())
+                .with(FileDatabaseHistory.FILE_PATH, DB_HISTORY_PATH)
+                .build();
```
Comment (rhauch) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlConnectorRegressionIT.java]: Again, reformatting changes like these are unnecessary and obscure the actual changes.

**Comment by omarsmak** [comment]
@rhauch Sorry for the formatting issue. I updated the PR with the correct changes :)

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    }
+
+    /**
+     * Convert original value insertion of type 'SMALLINT' into the correct SMALLINT UNSIGNED representation
+     * Note: Unsigned SMALLINT (16-bit) is represented in 'Integer' 32-bit data type. Reference: https://kafka.apache.org/0102/javadoc/org/apache/kafka/connect/data/Schema.Type.html
+     *
+     * @param originalNumber {@link Integer} the original insertion value
+     * @return {@link Integer} the correct representation of the original insertion value
+     */
+    public static Integer convertUnsignedSmallint(Integer originalNumber){
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java]: Hey @omarsmak, is there any specific reason for working with the wrapper type ( `Integer`) instead of its primitive counterpart (`int`)? I'm a bit concerned about the object allocations caused by this, e.g. by the calculation below but also by explicit constructor usage in some places.

Comment (omarsmak) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java]: Well, not an obvious reason, when I was looking for example at `JdbcValueConverters.java`, the wrapper type `Integer` was used to verify the types and I didn't want to break anything. Anyway, we have been using already a patched version of DBZ that include this fix internally and is all smooth :)

Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java]: Yes, I'm confident it works, but it could work a bit better actually :) I've just rebased and merged this PR, adding one more commit to avoid some of the auto-boxing. All tests pass and it should be alright, but if you could take another look at this last commit on master, that'd be great.

**Comment by gunnarmorling** [comment]
Rebased and applied. Thanks a lot, @omarsmak! I'm planning to do the 0.5.1 release with this change later today.

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java (hunk 3 above)
Comment (omarsmak) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java]: Awesome then, will do 👍
